### ISSUE TITLE
Visual Studio 17.11.5 causes Windows MAUI to break build

### ISSUE THREAD

**Issue by jayhayman-hdd** [dotnet/maui#25160]
### Description

Following on from issue: https://github.com/dotnet/maui/issues/25132

---

Since installing 17.11.5, all of our MAUI builds are failing for Windows with errors along the lines of:

```
The type 'CoreWebView2' exists in both 'Microsoft.Web.WebView2.Core, Version=1.0.864.35, Culture=neutral, PublicKeyToken=2a8ab48044d2601e' and 'Microsoft.WinUI, Version=3.0.0.0, Culture=neutral, PublicKeyToken=de31ebe4ad15742b'	obj\Debug\net8.0-windows10.0.19041.0\win10-x64\intermediatexaml\WinRT.SourceGenerator\Generator.RcwReflectionFallbackGenerator\RcwFallbackInitializer.g.cs
```
and
```
The type 'CoreWebView2HttpHeadersCollectionIterator' exists in both 'Microsoft.Web.WebView2.Core, Version=1.0.864.35, Culture=neutral, PublicKeyToken=2a8ab48044d2601e' and 'Microsoft.WinUI, Version=3.0.0.0, Culture=neutral, PublicKeyToken=de31ebe4ad15742b'	Cylch.Atp.Maui.App (net8.0-windows10.0.19041.0)	\obj\Debug\net8.0-windows10.0.19041.0\win10-x64\intermediatexaml\WinRT.SourceGenerator\Generator.RcwReflectionFallbackGenerator\RcwFallbackInitializer.g.cs
```

Project 1 is a large MAUI project built with .Net 7 MAUI project templates and upgraded with every new release.

Project 2 is a smaller MAUI project built a few months ago with .Net 8 MAUI project templates.

Both projects tested return the same amount of errors (33). with the same errors.
 
Neither of these projects are intentionally using `WebView2` components that we are aware of.

Both projects target:

```
<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);net8.0-windows10.0.19041.0</TargetFrameworks>
```

And platform:

```
<TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</TargetPlatformMinVersion>
```

Project 1 has to target Windows 10 to keep compatibility with our customer.

**What I've tried to fix this from suggestions in other posts**

Adding...

```
<PropertyGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
    <WindowsSdkPackageVersion>10.0.19041.38</WindowsSdkPackageVersion>
</PropertyGroup>
```

Does not resolve any of the errors.

Adding...

```
CsWinRTRcwFactoryFallbackGeneratorForceOptOut
```

Removes all errors and both projects compile and run.

However, I am now getting a debugging attachment error in Project 1 when running my Windows app that does not appear on machines that have not been upgraded to 17.11.5.

![Image](https://github.com/user-attachments/assets/8b50a879-0324-4e7e-b0f1-f83b19ce0366)

I cannot tell yet if this is an error related to the 17.11.5 update (did anything MAUI based get updated in 17.11.5?) or a new issue for us. I will spend more time on it tomorrow. iOS and Android work as expected, it only seems to be a Windows issue - break points still work but I've not been able to determine what line in our code is causing this. I will update as I debug.

Adding both `WindowsSdkPackageVersion` and `CsWinRTRcwFactoryFallbackGeneratorForceOptOut` only outputs `unsafe` errors:

```
Unsafe code may only appear if compiling with /unsafe	\obj\Debug\net8.0-windows10.0.19041.0\win10-x64\intermediatexaml\WinRT.SourceGenerator\Generator.WinRTAotSourceGenerator\WinRTGenericInstantiation.g.cs
```

When me and my team have some spare time, we'll try to rebuild a repo for testing.

### Version with bug

8.0.91 SR9.1

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

8.0.91 SR9.1

### Affected platforms

Windows

### Affected platform versions

net8.0-windows10.0.19041.0 / 10.0.17763.0

**Comment by Redth** [comment]
Is your project specifying explicit versions of WindowsAppSDK and/or WebView2 nuget packages?

**Comment by jayhayman-hdd** [comment]
`WindowsAppSDK  1.5.240802000`.

No NuGet reference to `WebView2`.

**Comment by Redth** [comment]
I'm not able to reproduce this so far.  Are you able to share a reproduction of it?

**Comment by jayhayman-hdd** [comment]
Sure no problem. I'll get someone on my team or me to sort this out ASAP.

**Comment by manodasanW** [comment]
For the `RcwReflectionFallbackGenerator` compiler issue, I am working on a fix in the source generator to address this and turn it into a warning instead, but I believe it only happens when there is either a direct or transitive dependency to an older version of the WebView2 package used together with WinAppSDK 1.5 which has the same WebView APIs.

**Comment by niksedk** [comment]
I have the same issue with this project: https://github.com/niksedk/subtitle-alchemist
(and other projects as well)

One computer where I updated it just crashes, and another computer with no visual studio update it works fine.

I'm on Windows 11

![Image](https://github.com/user-attachments/assets/57b4124e-2ebb-4575-b460-b8bb857ffcc4)



Event viewer:
````
Log Name:      Application
Source:        Application Error
Date:          09/10/2024 23.19.45
Event ID:      1000
Task Category: Application Crashing Events
Level:         Error
Keywords:      
User:          LAPTOP-JUSVDOJ0\nikse
Computer:      LAPTOP-JUSVDOJ0
Description:
Faulting application name: SubtitleAlchemist.exe, version: 1.0.0.0, time stamp: 0x66e80000
Faulting module name: CoreMessagingXP.dll, version: 10.0.26105.1010, time stamp: 0x3cb81cec
Exception code: 0xc000027b
Fault offset: 0x00000000000945b2
Faulting process id: 0x0x4550
Faulting application start time: 0x0x1DB1A90F11A77C6
Faulting application path: C:\git\subtitle-alchemist\src\bin\Debug\net8.0-windows10.0.19041.0\win10-x64\AppX\SubtitleAlchemist.exe
Faulting module path: C:\Program Files\WindowsApps\Microsoft.WindowsAppRuntime.1.5_5001.275.500.0_x64__8wekyb3d8bbwe\CoreMessagingXP.dll
Report Id: be9846d4-086d-46ad-8246-43034bfaee45
Faulting package full name: dk.nikse.subtitlealchemist_0.0.6.0_x64__f9j3vma71va8p
Faulting package-relative application ID: App
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" Guid="{a0e9b465-b939-57d7-b27d-95d8e925ff57}" />
    <EventID>1000</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>100</Task>
    <Opcode>0</Opcode>
    <Keywords>0x8000000000000000</Keywords>
    <TimeCreated SystemTime="2024-10-09T21:19:45.0432328Z" />
    <EventRecordID>338880</EventRecordID>
    <Correlation />
    <Execution ProcessID="4668" ThreadID="1612" />
    <Channel>Application</Channel>
    <Computer>LAPTOP-JUSVDOJ0</Computer>
    <Security UserID="S-1-5-21-2051407611-681702476-311258538-1001" />
  </System>
  <EventData>
    <Data Name="AppName">SubtitleAlchemist.exe</Data>
    <Data Name="AppVersion">1.0.0.0</Data>
    <Data Name="AppTimeStamp">66e80000</Data>
    <Data Name="ModuleName">CoreMessagingXP.dll</Data>
    <Data Name="ModuleVersion">10.0.26105.1010</Data>
    <Data Name="ModuleTimeStamp">3cb81cec</Data>
    <Data Name="ExceptionCode">c000027b</Data>
    <Data Name="FaultingOffset">00000000000945b2</Data>
    <Data Name="ProcessId">0x4550</Data>
    <Data Name="ProcessCreationTime">0x1db1a90f11a77c6</Data>
    <Data Name="AppPath">C:\git\subtitle-alchemist\src\bin\Debug\net8.0-windows10.0.19041.0\win10-x64\AppX\SubtitleAlchemist.exe</Data>
    <Data Name="ModulePath">C:\Program Files\WindowsApps\Microsoft.WindowsAppRuntime.1.5_5001.275.500.0_x64__8wekyb3d8bbwe\CoreMessagingXP.dll</Data>
    <Data Name="IntegratorReportId">be9846d4-086d-46ad-8246-43034bfaee45</Data>
    <Data Name="PackageFullName">dk.nikse.subtitlealchemist_0.0.6.0_x64__f9j3vma71va8p</Data>
    <Data Name="PackageRelativeAppId">App</Data>
  </EventData>
</Event>
````

**Comment by JoeTomkinson** [comment]
I am seeing the same issue with WinUI 3 projects. I am using the below due to migrating to .NET 8. This only started when I ran the VS2022 update.


<WindowsSdkPackageVersion>10.0.26100.38</WindowsSdkPackageVersion>

**Comment by jayhayman-hdd** [comment]
@niksedk - are you using Skia Sharp extended: `xmlns:controls="clr-namespace:SkiaSharp.Extended.UI.Controls;assembly=SkiaSharp.Extended.UI"` or anything like Lottie animations `<controls:SKLottieView...`?

If I disable these then my application progresses.

Not sure if this is an issue with the package or the fact it's loading resources from `Resources/Raw/Lottie`. More investigation today.

Previously when running our app in debug mode for either Android or Windows, we'd sometimes, maybe once every 100 build-runs, MAUI will hang on our loading screen as if the debugger hasn't attached or can't progress. Since 17.11.5, instead of hanging it now show the detached debugger modal as in my screen shot in the OP. This now happens every time I build, unless I uninstall the app from the device and debug run again.

This might be a separate issue, I'll try to get more information today.

For ref:

![Image](https://github.com/user-attachments/assets/5cc00bb0-ef34-4aff-9e30-38ba57de037d)

**Comment by niksedk** [comment]
I'm using this Skia:
![Image](https://github.com/user-attachments/assets/e90780fb-4cb8-424e-8621-5bc9242284b9)

**Comment by jayhayman-hdd** [comment]
@niksedk are any of your resources using Skia loaded or saved to `Resources/Raw`?

**Comment by niksedk** [comment]
I do not think so... but I'm not entirely sure - you can check the source code: https://github.com/niksedk/subtitle-alchemist

**Comment by manodasanW** [comment]
I just investigated another similar issue related to Skia which it seems both repros here have too.  See here for more details and a workaround until we fix it: https://github.com/dotnet/maui/issues/25150#issuecomment-2404428189

And the `RcwReflectionFallbackGenerator` issue with WebView will be fixed by https://github.com/microsoft/CsWinRT/pull/1816

**Comment by niksedk** [comment]
Thanks for the workaround, works.

**Comment by Zhanglirong-Winnie** [comment]
This issue has been verified using Visual Studio 17.11.5 (8.0.91 & 8.0.82 &8.0.70 ). Can repro on Windows platform. 
Unable to verify regression, an error will be reported if the version is lower than 8.0.71
https://github.com/niksedk/subtitle-alchemist

**Comment by jayhayman-hdd** [comment]
Can confirm adding

```
<CsWinRTRcwFactoryFallbackGeneratorForceOptOut>true</CsWinRTRcwFactoryFallbackGeneratorForceOptOut>
```

And

```
<PropertyGroup>
	<WindowsSdkPackageVersion>10.0.19041.34</WindowsSdkPackageVersion>
</PropertyGroup>
```

Has fixed the issue for now. However the SDK package didn't automatically update. On two machines, we did the following:

1. Added the above to `.proj` file.
2. Closed VS.
3. Clean build.
4. Deleted `.bin / .obj` folders.
5. Ran `dotnet restore`.
6. Opened VS and build,

Should we remove these elements from the `.proj` file when .Net 9 goes GA?

**Comment by Redth** [comment]
> For the `RcwReflectionFallbackGenerator` compiler issue, I am working on a fix in the source generator to address this and turn it into a warning instead, but I believe it only happens when there is either a direct or transitive dependency to an older version of the WebView2 package used together with WinAppSDK 1.5 which has the same WebView APIs.

That would make sense, and then I suspect some folks are hitting this using packages like `SkiaSharp.Views.Maui.Controls` which transitively depends on an older `Microsoft.WindowsAppSDK` which then depends on an older `Microsoft.Web.WebView2`.

**Comment by JORGEGO** [comment]
Im my case the exception produce when I call :  var location = await Geolocation.GetLocationAsync(request);
![Image](https://github.com/user-attachments/assets/d216a9af-cddc-4545-96ef-00278b64cef3)

The exception is producing in Windows
<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);net8.0-windows10.0.26100.0</TargetFrameworks>
And the event Viewer:

Faulting application name: greenyng.mobile.business.Operator.exe, version: 1.0.0.0, time stamp: 0x66e80000
Faulting module name: CoreMessagingXP.dll, version: 10.0.26105.1010, time stamp: 0x3cb81cec
Exception code: 0xc000027b
Fault offset: 0x00000000000945b2
Faulting process id: 0x0x8538
Faulting application start time: 0x0x1DB1B3BC52E9A3C
Faulting application path: C:\gm\greenyng.mobile\greenyng.mobile.business.Operator\bin\Debug\net8.0-windows10.0.26100.0\win10-x64\AppX\greenyng.mobile.business.Operator.exe
Faulting module path: C:\Program Files\WindowsApps\Microsoft.WindowsAppRuntime.1.5_5001.275.500.0_x64__8wekyb3d8bbwe\CoreMessagingXP.dll
Report Id: 80be37be-17c7-424e-b6ae-97663e2ca396
Faulting package full name: f52ad300-c749-4793-9424-d6d77b0cc588_1.0.0.1_x64__9zz4h110yvjzm
Faulting package-relative application ID: App

So my try catch not catch the exception!!
I have the las version of all Visual studio....

**Comment by sbloomztf** [comment]
I have a similar issue with my builds since 17.11.5

```
1>C:\Users\--\Mobile\xxx\obj\xxx\net8.0-windows10.0.19041.0\win10-x86\WinRT.SourceGenerator\Generator.WinRTAotSourceGenerator\WinRTGlobalVtableLookup.g.cs(5,21,5,38): error CS0116: A namespace cannot directly contain members such as fields, methods or statements
1>C:\Users\--\Mobile\xxx\obj\xxx\\net8.0-windows10.0.19041.0\win10-x86\WinRT.SourceGenerator\Generator.WinRTAotSourceGenerator\WinRTGlobalVtableLookup.g.cs(5,39,5,60): error CS0116: A namespace cannot directly contain members such as fields, methods or statements
1>C:\Users\--\Mobile\xxx\obj\xxx\\net8.0-windows10.0.19041.0\win10-x86\WinRT.SourceGenerator\Generator.WinRTAotSourceGenerator\WinRTGlobalVtableLookup.g.cs(5,16,5,21): error CS1514: { expected
1>C:\Users\--\Mobile\xxx\obj\xxx\\net8.0-windows10.0.19041.0\win10-x86\WinRT.SourceGenerator\Generator.WinRTAotSourceGenerator\WinRTGlobalVtableLookup.g.cs(5,38,5,39): error CS1022: Type or namespace definition, or end-of-file expected
1>C:\Users\--\Mobile\xxx\obj\xxx\\net8.0-windows10.0.19041.0\win10-x86\WinRT.SourceGenerator\Generator.WinRTAotSourceGenerator\WinRTGlobalVtableLookup.g.cs(6,1,6,2): error CS1022: Type or namespace definition, or end-of-file expected
```

I have multiple similar projects sharing the same "core" project, of these, 2 have these build errors, another runs fine...   I've tried a few of the mentioned work arounds without success so far.

**Comment by ARiSoul** [comment]
Same problem here, with not a single reference to any WebView, with no third party controls, nothing... A very simple project that I've started last week. After upgrading Visual Studio yesterday to 17.11.5, simply stoped working with multiple errors as described in the issue description.

Just to confirm, that simply adding <WindowsSdkPackageVersion>10.0.19041.34</WindowsSdkPackageVersion>, solved the issue.

Thank you for the workaround.

**Comment by JORGEGO** [comment]
I've added this line Like:
<TargetFrameworks>net8.0-ios;net8.0-maccatalyst;net8.0-android34.0</TargetFrameworks>
<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);net8.0-windows10.0.26100.0</TargetFrameworks>


<OutputType>Exe</OutputType>
<RootNamespace>greenyng.mobile.business.Operator</RootNamespace>
<UseMaui>true</UseMaui>
<SingleProject>true</SingleProject>
<ImplicitUsings>enable</ImplicitUsings>


<ApplicationTitle>greenyng.mobile.business.Operator</ApplicationTitle>


<ApplicationId>com.companyname.greenyng.mobile.business.operator</ApplicationId>
<ApplicationIdGuid>f52ad300-c749-4793-9424-d6d77b0cc588</ApplicationIdGuid>


<ApplicationDisplayVersion>1.0</ApplicationDisplayVersion>
<ApplicationVersion>1</ApplicationVersion>

<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">11.0</SupportedOSPlatformVersion>
<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">13.1</SupportedOSPlatformVersion>
<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">21.0</SupportedOSPlatformVersion>
<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</SupportedOSPlatformVersion>
<TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</TargetPlatformMinVersion>
<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'tizen'">6.5</SupportedOSPlatformVersion>

<WindowsSdkPackageVersion>10.0.19041.34</WindowsSdkPackageVersion>



But I receive this error compiling:
.nuget\packages\communitytoolkit.mvvm\8.3.2\buildTransitive\CommunityToolkit.Mvvm.WindowsSdk.targets(81,5): error MVVMTKCFG0003: This version of the MVVM Toolkit requires 'Microsoft.Windows.SDK.NET.Ref' version '10.0.19041.38' or later. Please update to .NET SDK 8.0.109, 8.0.305 or 8.0.402 (or later)

I Check which version I have installed:
C:\Program Files\Microsoft Visual Studio\2022\Professional>dotnet --list-sdks
7.0.401 [C:\Program Files\dotnet\sdk]
8.0.403 [C:\Program Files\dotnet\sdk]

So I use a later version...  but if I add this line I could not compile...

Any sugestion?

**Comment by JORGEGO** [comment]
now it works, In my case was the version of <PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.2" /> 
I had a bigger one, thanks to Copilot I was suggested to downgrade to this version
now it works

**Comment by JogyBlack** [comment]
The `<WindowsSdkPackageVersion>10.0.19041.34</WindowsSdkPackageVersion>` workaround unfortunately does not work for me, as I am using latest MVVM Toolkit which requires version 10.0.19041.38 or later.

The only way I was able to successfully compile my small program was by adding `<AllowUnsafeBlocks>True</AllowUnsafeBlocks>`

**Comment by reinierkops** [comment]
I'm facing the same problem. But I have also incorrect generated codefiles, for example WinRTGlobalVtableLookup.g.cs which cause build errors.

**Comment by stephenquan** [comment]
To use the latest CommuintyToolkit.Mvvm on Windows you may need to set your WindowsSdkPackageVersion to a .41 version, see https://github.com/CommunityToolkit/dotnet/issues/893#issuecomment-2316705291

**Comment by Sydney680928** [comment]
I have also incorrect generated codefiles, for example WinRTGlobalVtableLookup.g.cs which cause build errors.

**Comment by gdamino** [comment]
Same here, after update to vs 17.11.5 program crash at startup.
Any workarounds?

**Comment by spatialdude** [comment]
Strangely Visual Studio 2022 Preview 17.12.0 [Preview 3.0](https://learn.microsoft.com/en-au/visualstudio/releases/2022/release-notes-preview?tabs=preview3#tabpanel_1_preview3) appears to work ok. This hints at the problem being more to do with Visual Studio 17.11.5 than MAUI? 🤔

**Comment by juwens** [comment]
It’s exactly the same as it was with Xamarin.Forms. Don’t expect the same people (which are blind to the fundamental issues of the project) and have created half baked solutions for a decade, to come up with working solutions now. The only thing which changed is the name of the project, the codebase and quality-assurance is the same.


> Same here, after update to vs 17.11.5 program crash at startup. Any workarounds?
> 
> Edit:
> 
> As mentioned some post ago adding these lines to maui project resolved the issue:
> 
> `<CsWinRTRcwFactoryFallbackGeneratorForceOptOut>true</CsWinRTRcwFactoryFallbackGeneratorForceOptOut> <WindowsSdkPackageVersion>10.0.19041.34</WindowsSdkPackageVersion>`
> 
> But the problem is not resolved, some ms devs here have to figure out the problem and why MAUI projects stop to build every time Visual Studio is updated. I have a lot of "workarounds" on my project file...

**Comment by rarescho** [comment]
Microsoft is telling me to follow the issue here, but no one of them is giving answers as I see. 

Did someone of you found a solution?
Now I'm trying the @spatialdude way

**Comment by rarescho** [comment]
> Strangely Visual Studio 2022 Preview 17.12.0 [Preview 3.0](https://learn.microsoft.com/en-au/visualstudio/releases/2022/release-notes-preview?tabs=preview3#tabpanel_1_preview3) appears to work ok. This hints at the problem being more to do with Visual Studio 17.11.5 than MAUI? 🤔

Not working for me

**Comment by manodasanW** [comment]
@rarescho Can you confirm which error you are seeing?  Based on that, I should be able to provide a solution to get unblocked until the issue is fixed by default.

**Comment by Sydney680928** [comment]
CS1514	{ attendue	MY-COMETA (net8.0-windows10.0.19041.0)	D:\GITHUB\COMETA_APPS\MY-COMETA\obj\Debug\net8.0-windows10.0.19041.0\win10-x64\intermediatexaml\WinRT.SourceGenerator\Generator.WinRTAotSourceGenerator\WinRTGlobalVtableLookup.g.cs	5		
CS0116	Un espace de noms ne peut pas contenir directement des membres tels que des champs, des méthodes ou des instructions	MY-COMETA (net8.0-windows10.0.19041.0)	D:\GITHUB\COMETA_APPS\MY-COMETA\obj\Debug\net8.0-windows10.0.19041.0\win10-x64\intermediatexaml\WinRT.SourceGenerator\Generator.WinRTAotSourceGenerator\WinRTGlobalVtableLookup.g.cs	5		
CS1022	Définition de type ou d'espace de noms, ou fin de fichier attendue	MY-COMETA (net8.0-windows10.0.19041.0)	D:\GITHUB\COMETA_APPS\MY-COMETA\obj\Debug\net8.0-windows10.0.19041.0\win10-x64\intermediatexaml\WinRT.SourceGenerator\Generator.WinRTAotSourceGenerator\WinRTGlobalVtableLookup.g.cs	6

**Comment by eschoker** [comment]
I can reproduce

The issue came for me after I update windows and visual studio

![Image](https://github.com/user-attachments/assets/ad883c48-28b7-4eb9-8e7a-46f2f1424d40)

**Comment by juwens** [comment]
I made our project work with vs 17.11.5 by pinning SDK version to 8.0.402.  
(400 and 401 work too. That means the regression is in SDK 402 -> 403.)

* VS 17.11.4 is shipped with SDK 8.0.400
* VS 17.11.5 is shipped with SDK 8.0.403

global.json
```
{
  "sdk": {
    "version": "8.0.402",
    "rollForward": "disable"
  }
}
```

PS: the diff in the sdk seems unsuspicious: https://github.com/dotnet/sdk/compare/v8.0.403...v8.0.402

**Comment by juwens** [comment]
I dumped the stowed exception in windbg:

```
0:000> 
ExceptionAddress: 00007ffad4632f62 (KERNELBASE!RaiseFailFastException+0x0000000000000152)
   ExceptionCode: c000027b
  ExceptionFlags: 00000001
NumberParameters: 2
   Parameter[0]: 00000263f83dcd40
   Parameter[1]: 0000000000000001
0:000> 
ExceptionAddress: 00007ffad4632f62 (KERNELBASE!RaiseFailFastException+0x0000000000000152)
   ExceptionCode: c000027b
  ExceptionFlags: 00000001
NumberParameters: 2
   Parameter[0]: 00000263f83dcd40
   Parameter[1]: 0000000000000001
0:000> 
ExceptionAddress: 00007ffad4632f62 (KERNELBASE!RaiseFailFastException+0x0000000000000152)
   ExceptionCode: c000027b
  ExceptionFlags: 00000001
NumberParameters: 2
   Parameter[0]: 00000263f83dcd40
   Parameter[1]: 0000000000000001
0:000> !pde.dse
The call to LoadLibrary(pde) failed, Win32 error 0n2
    "The system cannot find the file specified."
Please check your debugger configuration and/or network access.
Error: Failed to load extension pde
0:000> !pde.dse
=========================================================================================
 PDE v11.3 - Copyright 2017 Andrew Richards
=========================================================================================
Stowed Exception Array @ 0x00000263f83dcd40

Stowed Exception #1 @ 0x00000263f1e4a228
	0x80131604 (FACILITY_URT - .NET CLR): Uncaught exception thrown by method called through Reflection.

	Stack	 : 0x263f83dc4b0
		7ffad4ec9da4 combase!RoOriginateLanguageException+0x54
		7ff9a4ad7fd1

>>> Associated CLR Exception <<<

Exception object: 00000223835c1728
Exception type:   System.Reflection.TargetInvocationException
Message:          Exception has been thrown by the target of an invocation.
InnerException:   System.NotSupportedException, Use !PrintException 00000223835C1510 to see more.
StackTrace (generated):
    SP               IP               Function
    000000DE9A9EA480 00007FF9A4092FFF System_Private_CoreLib!System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(System.Object, System.Reflection.BindingFlags)+0x15f
    000000DE9A9ECF90 00007FF9FAD824E2 System_Private_CoreLib!System.Reflection.MethodBase.Invoke(System.Object, System.Object[])+0x22
    000000DE9A9ECFD0 00007FF9A4AA396E WinRT_Runtime!WinRT.IWinRTObject.<IsInterfaceImplementedFallback>g__GetObjectReferenceViaVftbl|1_1(WinRT.IObjectReference, System.Type)+0x9e
    000000DE9A9ED020 00007FF9A332A7E4 WinRT_Runtime!WinRT.IWinRTObject.IsInterfaceImplementedFallback(System.RuntimeTypeHandle, Boolean)+0x1074
    000000DE9A9ED3F0 00007FF9A33296FD WinRT_Runtime!WinRT.IWinRTObject.System.Runtime.InteropServices.IDynamicInterfaceCastable.IsInterfaceImplemented(System.RuntimeTypeHandle, Boolean)+0x3d
    000000DE9A9ED420 00007FF9FAD62D94 System_Private_CoreLib!System.Runtime.InteropServices.DynamicInterfaceCastableHelpers.IsInterfaceImplemented(System.Runtime.InteropServices.IDynamicInterfaceCastable, System.RuntimeType, Boolean)+0x24
    000000DE9A9ED7E0 00007FF9A4A9E600 WinRT_Runtime!WinRT.CastExtensions.As[[System.__Canon, System.Private.CoreLib]](System.Object)+0xe0
    000000DE9A9ED860 00007FF9A4AD7610 SkiaSharp_Views_Windows!SkiaSharp.Views.Windows.Utils.GetByteBuffer(Windows.Storage.Streams.IBuffer)+0x30
    000000DE9A9ED8B0 00007FF9A4AD7452 SkiaSharp_Views_Windows!SkiaSharp.Views.Windows.WindowsExtensions.GetPixels(Microsoft.UI.Xaml.Media.Imaging.WriteableBitmap)+0x22
    000000DE9A9ED8E0 00007FF9A4AD650A SkiaSharp_Views_Windows!SkiaSharp.Views.Windows.SKXamlCanvas.CreateBitmap(SkiaSharp.SKSizeI ByRef, Single ByRef)+0x2da
    000000DE9A9ED9E0 00007FF9A4AD5FB1 SkiaSharp_Views_Windows!SkiaSharp.Views.Windows.SKXamlCanvas.DoInvalidate()+0x91
    000000DE9A9EDB10 00007FF9A4AD1189 Microsoft_InteractiveExperiences_Projection!ABI.Microsoft.UI.Dispatching.DispatcherQueueHandler.Do_Abi_Invoke(IntPtr)+0x39

StackTraceString: <none>
HResult: 80131604

0:000> !PrintException /d 00000223835C1510
Exception object: 00000223835c1510
Exception type:   System.NotSupportedException
Message:          Managed vtable types (ie. containing any reference types) are not supported.
InnerException:   <none>
StackTrace (generated):
    SP               IP               Function
    000000DE9A9EC8A0 00007FF9A4AD7BF9 WinRT_Runtime!WinRT.ObjectReference`1[[ABI.SkiaSharp.Views.Windows.IBufferByteAccess+Vftbl, SkiaSharp.Views.Windows]].GetVtable(IntPtr)+0x69
    000000DE9A9EC8F0 00007FF9A4AD7B40 WinRT_Runtime!WinRT.ObjectReference`1[[ABI.SkiaSharp.Views.Windows.IBufferByteAccess+Vftbl, SkiaSharp.Views.Windows]]..ctor(IntPtr)+0x40
    000000DE9A9EC970 00007FF9A4AD7A1F WinRT_Runtime!WinRT.ObjectReference`1[[ABI.SkiaSharp.Views.Windows.IBufferByteAccess+Vftbl, SkiaSharp.Views.Windows]].Attach(IntPtr ByRef, System.Guid)+0x8f
    000000DE9A9ECA20 00007FF9A4AD78E2 WinRT_Runtime!WinRT.ObjectReference`1[[ABI.SkiaSharp.Views.Windows.IBufferByteAccess+Vftbl, SkiaSharp.Views.Windows]].TryAs(WinRT.IObjectReference, System.Guid, WinRT.ObjectReference`1<Vftbl> ByRef)+0xc2
    000000DE9A9ECAB0 00007FF9A4AD7800 WinRT_Runtime!WinRT.IObjectReference.TryAs[[ABI.SkiaSharp.Views.Windows.IBufferByteAccess+Vftbl, SkiaSharp.Views.Windows]](System.Guid, WinRT.ObjectReference`1<Vftbl> ByRef)+0x90
    000000DE9A9ECB10 00007FF9A4AD7745 WinRT_Runtime!WinRT.IObjectReference.As[[ABI.SkiaSharp.Views.Windows.IBufferByteAccess+Vftbl, SkiaSharp.Views.Windows]](System.Guid)+0x45
    000000DE9A9ECB60 00007FF9A4AD76D7 WinRT_Runtime!WinRT.IObjectReference.As[[ABI.SkiaSharp.Views.Windows.IBufferByteAccess+Vftbl, SkiaSharp.Views.Windows]]()+0x47
    000000DE9A9ECF30 0000000000000000 System_Private_CoreLib!System.RuntimeMethodHandle.InvokeMethod(System.Object, Void**, System.Signature, Boolean)+0x10ea310
    000000DE9A9ECF30 00007FF9A4092EEB System_Private_CoreLib!System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(System.Object, System.Reflection.BindingFlags)+0x4b

StackTraceString: <none>
HResult: 80131515

```

**Comment by manodasanW** [comment]
Looking through the last couple issues mentioned (RcwReflectionFallbackGenerator, skia, WinRTGlobalVtableLookup), they look like the ones that we have addressed for the upcoming .NET SDK update.  If you want to try out the fix earlier, you can specify the `WindowsSdkPackageVersion` property in your `csproj` or in a common props file like `Directory.Build.Props` to one of the versions mentioned [here](https://github.com/microsoft/CsWinRT/releases/tag/2.1.6.241013.1) based on version of .NET that you target.  You may need to delete bin / obj for it to pick it up.  If you still run into issues with the new versions, let me know and I can look into it.

**Comment by Sydney680928** [comment]
If I use <WindowsSdkPackageVersion>10.0.19041.54</WindowsSdkPackageVersion>it's ok !

**Comment by SimonMeadLexacom** [comment]
I also encountered this issue a few days ago. Adding CsWinRTRcwFactoryFallbackGeneratorForceOptOut as mentioned above is working as a workaround for me
`<CsWinRTRcwFactoryFallbackGeneratorForceOptOut>true</CsWinRTRcwFactoryFallbackGeneratorForceOptOut>`

**Comment by aloh86** [comment]
> If I use WindowsSdkPackageVersion with 10.0.19041.54 it's ok !

This works for me for a broken WinUI 3 project after upgrading to 17.11.5. How did you find the solution and why does this work?

**Comment by mattleibow** [comment]
There is now a new SkiaSharp build out on nuget that should be fixed!

**Comment by nefen** [comment]
By adding this:
  <PropertyGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
    <WindowsSdkPackageVersion>10.0.19041.52</WindowsSdkPackageVersion>
  </PropertyGroup>

I managed to compile.

**Comment by stephenquan** [comment]
17.12.0 is out, on initial testing, it looks good.
